### ISSUE TITLE
Generator function causing JS_SYNTAX_ERROR 

### ISSUE THREAD

**Issue by lidel** [mozilla/addons-linter#2108]
### Describe the problem and steps to reproduce it:

Attempted to validate an extension whose background .js had [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*):

```js
content: (async function * () { 
// ...
```

#### How to reproduce

```bash
git clone https://github.com/mozilla/libdweb.git 
cd libdweb && git checkout fbff435b33542b02bc35c0bb8c27bc203e6d4e6e
cd demo/protocol && npx web-ext@2.7.0 lint
```

### What happened?
```
Code              Message                       Description
JS_SYNTAX_ERROR   JavaScript syntax error       There is a JavaScript syntax error in your code; validation cannot continue on this file.
```

### What did you expect to happen?

Firefox runs generators just fine. It guess [`function *`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*) should be accepted?
I may be missing some background on this, but MDN support table is green:

[![screenshot_11](https://user-images.githubusercontent.com/157609/42838129-3846a896-8a00-11e8-8ae4-5454a2e6af7d.png)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*#Browser_compatibility)

### Anything else we should know?

Versions:

```js
"web-ext": "2.7.0" with "addons-linter": "1.0.0"
```

Examples of use of generator functions in browser extensions:

- https://github.com/mozilla/libdweb/blob/fbff435b33542b02bc35c0bb8c27bc203e6d4e6e/demo/protocol/protocol.js#L6
- https://github.com/ipfs-shipyard/ipfs-companion/blob/08ef6f3866eb26e8340bb32249a8ccc8ca89c536/add-on/src/lib/ipfs-protocol-libdweb.js#L23

**Comment by lidel** [comment]
Hm.. I repeated the test wih latest version and the error is gone:

```bash
git clone https://github.com/mozilla/libdweb.git 
cd libdweb && git checkout fbff435b33542b02bc35c0bb8c27bc203e6d4e6e
cd demo/protocol && npx addons-linter@1.2.4 .
```

Does it mean generators were added after v1.0.0  and `web-ext` just needs to be bumped to the latest version of addons-linter?

**Comment by EnTeQuAk** [comment]
> Does it mean generators were added after 1.0.0 and web-ext just needs to be bumped to the latest version of addons-linter?

Correct. There were quite a few updates after 1.0.0 even that added support for many more recent JS features (e.g generators like you described, `catch` without an exception etc).

I'm closing this in favor of https://github.com/mozilla/web-ext/pull/1326
